Fix Bugs Introduced by #845 and QuadTree

lzardy opened the pull request:
1. Fixes vision issues
2. Fixes collision issues, specifically ones regarding buildings

lzardy committed:
Fix Bugs Introduced by #845 and QuadTree

1. Fixes vision issues
2. Fixes collision issues, specifically ones regarding buildings

lzardy commented:
I'd like to note that QuadTree seems to be a volatile addition in its current state.

compsi627 commented on GameServerLib/Content/NavGrid.cs:
```diff
@@ -544,7 +544,7 @@ public bool IsAnythingBetween(Vector2 a, Vector2 b)
 
         public bool IsAnythingBetween(IGameObject a, IGameObject b)
         {
-            var rayDist = Math.Sqrt(CastRaySqr(a.GetPosition(), b.GetPosition()));
```
Why is that needed? Also, I think that will break the lines come after.

danil179 replied:
Why is that needed? Also, I think that will break the lines come after.

danil179 commented:
> 
> 
> I'd like to note that QuadTree seems to be a volatile addition in its current state.

I'm not sure. The main problem is that the current lib of this QuadTree seems to be very unreliable, but widely used. I don't know if we do something wrong, or it's just the lib.

lzardy committed:
Small Fixes

Fixed IsAnythingBetween to use real rayDist and separated returns for buildings from other units.
Fixed CollisionHandler allowing collisions between a unit and itself.
Modified inhib and nexus radii to be more accurate (guess-work still).

danil179 commented on GameServerLib/Content/NavGrid.cs:
```diff
@@ -544,16 +544,20 @@ public bool IsAnythingBetween(Vector2 a, Vector2 b)
             {
+                var rayDist = Math.Sqrt(CastRaySqr(b.GetPosition(), a.GetPosition()));
                 rayDist += a.CollisionRadius;
+                return (rayDist * rayDist) < (b.GetPosition() - a.GetPosition()).SqrLength();
```
Take this line out of the if.

danil179 commented on GameServerLib/Content/NavGrid.cs:
```diff
@@ -544,16 +544,20 @@ public bool IsAnythingBetween(Vector2 a, Vector2 b)
+                var rayDist = Math.Sqrt(CastRaySqr(a.GetPosition(), b.GetPosition()));
+                rayDist = Math.Sqrt(CastRaySqr(a.GetPosition(), b.GetPosition()));
                 rayDist += b.CollisionRadius;
+                return (rayDist * rayDist) < (b.GetPosition() - a.GetPosition()).SqrLength();
```
And this. (Delete one of them)

danil179 reviewed (changes requested):
Apart from one comment, it looks good.

danil179 reviewed (approved):
Approved.
